buffer/src/main/java/io/netty/buffer/UnpooledByteBufAllocator.java

```diff
@@ -22,20 +22,36 @@
  */
 public final class UnpooledByteBufAllocator extends AbstractByteBufAllocator {
 
+    private final boolean disableLeakDetector;
+
     /**
-     * Default instance
+     * Default instance which uses leak-detection for direct buffers.
      */
     public static final UnpooledByteBufAllocator DEFAULT =
             new UnpooledByteBufAllocator(PlatformDependent.directBufferPreferred());
 
     /**
-     * Create a new instance
+     * Create a new instance which uses leak-detection for direct buffers.
      *
      * @param preferDirect {@code true} if {@link #buffer(int)} should try to allocate a direct buffer rather than
      *                     a heap buffer
      */
     public UnpooledByteBufAllocator(boolean preferDirect) {
+        this(preferDirect, false);
+    }
+
+    /**
+     * Create a new instance
+     *
+     * @param preferDirect {@code true} if {@link #buffer(int)} should try to allocate a direct buffer rather than
+     *                     a heap buffer
+     * @param disableLeakDetector {@code true} if the leak-detection should be disabled completely for this
+     *                            allocator. This can be useful if the user just want to depend on the GC to handle
+     *                            direct buffers when not explicit released.
+     */
+    public UnpooledByteBufAllocator(boolean preferDirect, boolean disableLeakDetector) {
```

Comment:
Instead of adding another `boolean` parameter, could we use `ResourceLeakDetector.Level` instead here? It will require some refactoring but it will give us the opportunity to apply different leak detection level for different allocators.